macOS/DuckDuckGo/Application/AppDelegate.swift

```diff
@@ -827,10 +827,16 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
                                                              keyValueStore: keyValueStore,
                                                              sessionRestorePromptCoordinator: sessionRestorePromptCoordinator,
                                                              pixelFiring: PixelKit.shared)
-
+#if APPSTORE
+        if AppVersion.runType != .uiTests {
+            updateController = AppStoreUpdateController()
+        }
+#endif
 #if SPARKLE
         if AppVersion.runType != .uiTests {
-            updateController = UpdateController(internalUserDecider: internalUserDecider)
+            updateController = SparkleUpdateController(internalUserDecider: internalUserDecider)
+
+            guard let updateController = updateController as? SparkleUpdateController else { return }
```

Comment:
The guard is a bit confusing, since we just initialized a `SparkleUpdateController` there should be no legit case for returning here.

If what we need is the cast, I'd suggest using something like this instead.

```suggestion
            let updateController = SparkleUpdateController(internalUserDecider: internalUserDecider)
            self.updateController = updateController
```